pinot-spi/src/main/java/org/apache/pinot/spi/data/readers/PrimaryKey.java

```diff
@@ -35,6 +36,10 @@ public PrimaryKey(Object[] values) {
     return _values;
   }
 
+  public byte[] asBytes() {
+    return SerializationUtils.serialize(_values);
```

Comment:
I think this util uses the java ser/de to serialize the values. Can it serialize the `ByteArray` values?
Java ser/de is usually not efficient, so we might want to build our own custom serializer.

Reply:
any issue with using java ser/de? I thought the pk is of primitive values, so is there a need to build a custom serializer?

Reply:
Primary key might contain `ByteArray` which is not `Serializable`.
Can we check the byte size for the java ser/de result? My impression is that it will be quite large. This will be a very frequent operation for upsert (per record), so want to make sure it is efficient

Reply:
oh, no, this is from Apache commons lib which can serialize any object. Let me add a test case on ByteArray as an example.

Reply:
I checked the internal implementation, and it is leveraging the `ObjectOutputStream` which uses the java ser/de.
Yeah, let's add a test to make sure it works as expected

Reply:
you are right. i need to mark ByteArray as serializable. Any other types that we need to consider?

Reply:
Other types should be fine